src/diffusers/models/attention_processor.py

```diff
@@ -1778,7 +1778,28 @@ def __call__(
             query = apply_rotary_emb(query, image_rotary_emb)
             key = apply_rotary_emb(key, image_rotary_emb)
 
-        hidden_states = F.scaled_dot_product_attention(query, key, value, dropout_p=0.0, is_causal=False)
+        if is_torch_npu_available():
```

Comment:
should we make a new attenton processor instead?

Reply:
Thanks for the recommendation, I've modified and make the new attention processor